jobs/JGDAS_ENKF_SELECT_OBS

```diff
@@ -51,18 +51,18 @@ mkdir -p "${COMOUT_ATMOS_ANALYSIS}"
 
 export ATMGES_ENSMEAN="${COMIN_ATMOS_HISTORY_PREV}/${GPREFIX}atmf006${GSUFFIX}"
 if [[ ! -f ${ATMGES_ENSMEAN} ]]; then
-    echo "FATAL ERROR: FILE MISSING: ATMGES_ENSMEAN = ${ATMGES_ENSMEAN}"
-    exit 1
+    export err=1
+    err_chk "FATAL ERROR: FILE MISSING: ATMGES_ENSMEAN = ${ATMGES_ENSMEAN}"
 fi
 
-# Ignore masking of chained commands and possible misspelling warning
-# shellcheck disable=SC2153,SC2312
+set +eu
+set -o pipefail
```

Comment:
Why is this and then `set +o pipefail / set_strict` needed?

Reply:
The `set -o pipefail` tells bash to return a non-zero status if any command in a piped set of commands fails.  E.g. `a=$(echo "hello" | grep "world" | echo "$1"`)` would return a non-zero status because the `grep` did not find a matching `"world"` string.  With `set +e`, this prevents the script from exiting out and instead just return a non-zero status, which can be captured with `export err=$?; err_chk`.  The `set_strict` call then runs `set -eu` so quick killing can be reenabled.

Reply:
Got it, thanks for that explanation!